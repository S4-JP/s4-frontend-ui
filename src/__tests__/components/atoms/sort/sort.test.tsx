import React from 'react'
import { render } from '@testing-library/react'
import { Sort, Color } from '../../../../index'

/**
 *   children?: string
 *   focus?: boolean
 *   sort?: number
 *   handleClick?(e: React.MouseEvent<HTMLElement>): void
 */
describe('Sort のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<Sort />)
    expect(getByTestId('atoms-s-wrap')).toMatchSnapshot()
  })

  test('childを設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(<Sort>テスト</Sort>)
    expect(getByTestId('atoms-s-wrap')).toMatchSnapshot()
  })

  test('props.sort=1設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(<Sort sort={1} />)

    expect(getByTestId('atoms-icon-svg-sort-asc')).toHaveStyle(
      `fill: ${Color.TEXT.LINK};`
    )
  })
  test('props.sort=2設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(<Sort sort={2} />)

    expect(getByTestId('atoms-icon-svg-sort-desc')).toHaveStyle(
      `fill: ${Color.TEXT.LINK};`
    )
  })
})
