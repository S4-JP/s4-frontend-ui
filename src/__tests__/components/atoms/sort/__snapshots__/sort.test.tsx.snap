// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort のレンダリング確認 childを設定した場合、設定値でレンダリングされる。 1`] = `
<div
  class="sc-bUbCnL cqBQSt"
  data-testid="atoms-s-wrap"
>
  <div
    class="sc-kIKDeO bOiMvG"
    data-testid="atoms-s-button-wrap"
  >
    <span
      class="sc-papXJ cwCWPm"
      color="#C6D6E3"
      data-testid="atoms-i-icon-wrap"
      size="8"
    >
      <svg
        data-testid="atoms-icon-svg-sort-asc"
        viewBox="0 0 20 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 137h40v40H32z"
          fill="none"
        />
        <path
          d="M19.421 10.522 12.2 1.088a2.772 2.772 0 0 0-4.4 0L.579 10.522A2.783 2.783 0 0 0 2.78 15h14.438a2.784 2.784 0 0 0 2.2-4.478"
        />
      </svg>
    </span>
    <span
      class="sc-papXJ cwCWPm"
      color="#C6D6E3"
      data-testid="atoms-i-icon-wrap"
      size="8"
    >
      <svg
        data-testid="atoms-icon-svg-sort-desc"
        viewBox="0 0 20 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 137h40v40H32z"
          fill="none"
        />
        <path
          d="M.579 4.478 7.8 13.912a2.772 2.772 0 0 0 4.4 0l7.218-9.434A2.783 2.783 0 0 0 17.218 0H2.782a2.784 2.784 0 0 0-2.2 4.478"
        />
      </svg>
    </span>
  </div>
  <div
    class="sc-hNKHps cJYfgR"
    data-testid="atoms-s-label"
  >
    テスト
  </div>
</div>
`;

exports[`Sort のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-bUbCnL cqBQSt"
  data-testid="atoms-s-wrap"
>
  <div
    class="sc-kIKDeO bOiMvG"
    data-testid="atoms-s-button-wrap"
  >
    <span
      class="sc-papXJ cwCWPm"
      color="#C6D6E3"
      data-testid="atoms-i-icon-wrap"
      size="8"
    >
      <svg
        data-testid="atoms-icon-svg-sort-asc"
        viewBox="0 0 20 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 137h40v40H32z"
          fill="none"
        />
        <path
          d="M19.421 10.522 12.2 1.088a2.772 2.772 0 0 0-4.4 0L.579 10.522A2.783 2.783 0 0 0 2.78 15h14.438a2.784 2.784 0 0 0 2.2-4.478"
        />
      </svg>
    </span>
    <span
      class="sc-papXJ cwCWPm"
      color="#C6D6E3"
      data-testid="atoms-i-icon-wrap"
      size="8"
    >
      <svg
        data-testid="atoms-icon-svg-sort-desc"
        viewBox="0 0 20 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 137h40v40H32z"
          fill="none"
        />
        <path
          d="M.579 4.478 7.8 13.912a2.772 2.772 0 0 0 4.4 0l7.218-9.434A2.783 2.783 0 0 0 17.218 0H2.782a2.784 2.784 0 0 0-2.2 4.478"
        />
      </svg>
    </span>
  </div>
</div>
`;
